config/default-config.yml

```diff
@@ -146,6 +146,9 @@ general:
   # GitHub token to use from composer
   #github_token: xxxxxx
 
+  # Set the default version of PHP CLI
+  # php_cli_version: 8.2
```

Comment:
note that by itself this would lead to 8.2 being installed then VVV failing once it restored PHP default as the newly set version wouldn't be installed. It also wouldn't change the dashboard and other sites which would still be configured to 8.2 as the default upstream. There's a lot more work needed to support this, and previously we've added the `php: ...` parameter to sites to make it easier to wrangle since most requests to change the default PHP were to fix the provisioner and CLI using the wrong PHP to run stuff which `php:..` fixes

Reply:
that is valid unintended effect that i didn't foresee. so i think its better to revert this change as it could break things rather than do good.

Reply:
I get why people want it, but I want to make sure it's not an X Y solution as it'll get abused and has lots of gotchas, I'm curious about your use case though

Reply:
I'm also mindful we will want a generic PHP provisioner at some point with templated PHP ini files etc rather than a dedicated hardcoded provisioner for each

Reply:
my use case maybe not that generic and/or critical enough, but here it is
- im trying to match php version in WP CLI to the one that i have on my production server, that way when i do test my script locally ( in vvv ), it ensure script run successfully in that specific php version. Changing PHP version for WP CLI in runtime is not that straightforward, so having the PHP CLI to be statically same with my production server, would avoid me make a wrong judgement.

the solution that i offered here, maybe just too far and/or not actually the solution that i actually need. in my other fork, i have this script in homebin: `vvv_change_php_default` and that was enough.
```php
#!/bin/bash

DEFAULTPHP=${1:-"7.4"}
echo " * Changing the default PHP CLI version ( ${DEFAULTPHP} )"
update-alternatives --set php "/usr/bin/php${DEFAULTPHP}"
update-alternatives --set phar "/usr/bin/phar${DEFAULTPHP}"
update-alternatives --set phar.phar "/usr/bin/phar.phar${DEFAULTPHP}"
update-alternatives --set phpize "/usr/bin/phpize${DEFAULTPHP}"
update-alternatives --set php-config "/usr/bin/php-config${DEFAULTPHP}"
echo " * Change completed"
```
i just assumed since its similar code with `vvv_restore_php_default`, then maybe having it in that script would be better, which is wrong assumption obviously.

Reply:
hmmm, something I wanted to do was make it auto-switch PHP based on the folder you were in, but it's not as straightforward especially if you've multiple sessions. There's a ticket to add the PHP version into the prompt so it's more obvious for people who aren't aware of all of this and mistakenly run in the wrong version they expect

Reply:
this also reverted, as you've mentioned that it can cause issues down the road. and my use case is too specific to be having this in default config